cc_plugin_og/checker.py

```diff
@@ -7,36 +7,47 @@
 class OGChecker(OGChecker):
     _cc_spec_version = "1.0"
     _cc_description = f"og {_cc_spec_version} compliance-checker"
-    _cc_display_headers = {3: "Highly Recommended", 2: "Recommended", 1: "Suggested"}
+    _cc_display_headers = {3: "Mandatory", 2: "Highly Recommended", 1: "Suggested"}
 
     METHODS_REGEX = re.compile(r"(\w+: *\w+) \((\w+: *\w+)\) *")
     PADDING_TYPES = ["none", "low", "high", "both"]
 
     def __init__(self):
         pass
 
-    def check_basic_requirements(self, dataset):
+    def check_mandatory_variables(self, dataset):
         """
         Check check_basic_requirements()
         """
 
         level = BaseCheck.HIGH
         score = 0
-        out_of = 1
         messages = []
         desc = "This checks basic requirements."
 
-        # Do test things here
-        ans = 0
-
-        try:
-            assert ans == 0
-        except AssertionError:
-            # Many messages may be appended here
-            m = '... test failed.'
-            messages.append(m)
-        else:
-            score += 1
+        required_variables = [
+            "LATITUDE_GPS",
+            "LONGITUDE_GPS",
+            "TIME_GPS",
+            "LATITUDE",
+            "LONGITUDE",
+            "TIME",
+            "TRAJECTORY",
+            "PLATFORM_MODEL",
+            "WMO_IDENTIFIER",
+            "PLATFORM_SERIAL_NUMBER",
+            "DEPLOYMENT_TIME",
+            "DEPLOYMENT_LATITUDE",
+            "DEPLOYMENT_LONGITUDE",
+        ]
+
+        out_of = len(required_variables)
+        for variable in required_variables:
+            try:
+                assert variable in dataset.variables
```

Comment:
We should not use Python `assert` outside of tests. It is dropped by JITs and other kinds of compilations. See https://docs.astral.sh/ruff/rules/assert/ for the lint explanation and some suggestions on how to replace it. Here we can do `if variable in dataset.variables: else ...`

PS: @benjwadams probably has a better way for these checks though, using existing compliance-checker machinery.

Reply:
`TestCtx` has methods that check for a particular named section and automatically adds the appropriate scoring values and messages depending on whether a test passes or fails: https://github.com/ioos/compliance-checker/blob/f543e4fa169635e31582b3804a7be06efa398db3/compliance_checker/base.py#L301-L358

I may rename it at some time as I'm using more Flask for some web applications and it has a "Test Context" that's used for unit tests.